VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Strings/StringDotFormatWithDifferentAmountOfArguments/StringDotFormatWithDifferentAmountOfArgumentsAnalyzer.cs

```diff
@@ -0,0 +1,150 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.Strings.StringDotFormatWithDifferentAmountOfArguments
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class StringDotFormatWithDifferentAmountOfArgumentsAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+        private static readonly string Category = VSDiagnosticsResources.StringsCategory;
+        private static readonly string Message = VSDiagnosticsResources.StringDotFormatWithDifferentAmountOfArgumentsMessage;
+        private static readonly string Title = VSDiagnosticsResources.StringDotFormatWithDifferentAmountOfArgumentsTitle;
+
+        internal static DiagnosticDescriptor Rule =>
+            new DiagnosticDescriptor(DiagnosticId.StringDotFormatWithDifferentAmountOfArguments, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.InvocationExpression);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = context.Node as InvocationExpressionSyntax;
+            if (invocation == null)
+            {
+                return;
+            }
+
+            if (invocation.ArgumentList == null)
+            {
+                return;
+            }
+
+            // Get the format string
+            // This corresponds to the argument passed to the parameter with name 'format'
+            var invokedMethod = context.SemanticModel.GetSymbolInfo(invocation);
+            var methodSymbol = invokedMethod.Symbol as IMethodSymbol;
+
+            // Verify we're dealing with a call to a method that accepts a variable named 'format' and a object, params object[] or a plain object[]
+            // params object[] and object[] can both be verified by looking for the latter
+            // This allows us to support similar calls like Console.WriteLine("{0}", "test") as well which carry an implicit string.Format
+            var formatParam = methodSymbol?.Parameters.FirstOrDefault(x => x.Name == "format");
+            if (formatParam == null)
+            {
+                return;
+            }
+
+            var hasObjectArray = 
+                methodSymbol.Parameters.FirstOrDefault(x => x.Type.Kind == SymbolKind.ArrayType && 
+                                                            ((IArrayTypeSymbol) x.Type).ElementType.SpecialType == SpecialType.System_Object) != null;
+            var hasObject = 
+                methodSymbol.Parameters.FirstOrDefault(x => x.Type.SpecialType == SpecialType.System_Object) != null;
+
+            if (!(hasObject || hasObjectArray))
```

Comment:
To be safer, you should check that _all_ remaining parameters (`.Skip(formatParam.Ordinal).All(...)`) are either `object` or `params object[]`